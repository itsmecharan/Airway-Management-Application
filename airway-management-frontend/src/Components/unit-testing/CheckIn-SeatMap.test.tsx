import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import * as reactRedux from "react-redux";
import configureMockStore from "redux-mock-store";
import CheckInSeatMap from "../check-in/CheckIn-SeatMap";
import Adapter from "@wojtekmaj/enzyme-adapter-react-17";
import Enzyme, { shallow, mount } from "enzyme";

Enzyme.configure({ adapter: new Adapter() });

const useSelectorMock = jest.spyOn(reactRedux, "useSelector");
const mockStore = configureMockStore();

const props = {
  checkin: {
    checkInSelectedFlightId: "",
    checkInFlightSchedule: [],
  },
};

const store = mockStore({ ...props });
const wrapper = mount(
  <Provider store={store}>
    <CheckInSeatMap />
  </Provider>
);

describe("Check In Seat Map Component Mounted", () => {
  beforeEach(() => {
    const {
      checkin: { checkInSelectedFlightId, checkInFlightSchedule },
    } = props;
    useSelectorMock.mockReturnValue({
      checkInSelectedFlightId,
      checkInFlightSchedule,
    });
  });
  afterAll(() => {
    useSelectorMock.mockClear();
  });
  it("It Should render", () => {
    expect(wrapper.exists()).toBe(true);
  });
  it("Verifying Dom Elements", () => {
    expect(wrapper.find("div").length).toBe(5);
    expect(wrapper.find("div div").length).toBe(4);
    expect(wrapper.find("div div button").length).toBe(5);
  });
  it("Creates a snapshot", () => {
    const rtrWrapper = renderer
      .create(
        <Provider store={store}>
          <CheckInSeatMap />
        </Provider>
      )
      .toJSON();
    expect(rtrWrapper).toMatchSnapshot();
  });
});
