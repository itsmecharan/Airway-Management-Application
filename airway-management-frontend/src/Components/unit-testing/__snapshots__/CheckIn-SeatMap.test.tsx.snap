// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check In Seat Map Component Mounted Creates a snapshot 1`] = `
<div>
  <div
    className="div-heading"
  >
    Passengers Seat Map
  </div>
  <div
    className="div-seatmap"
  />
  <div
    className="div-seatmap-btn-mark-area"
  >
    <button
      className="div-seatmap-btn-mark"
      disabled={true}
      style={
        Object {
          "background": "#43BE5F",
        }
      }
    />
     : 
    Checked In
     
    <br />
    <button
      className="div-seatmap-btn-mark"
      disabled={true}
      style={
        Object {
          "background": "white",
        }
      }
    />
     : 
    Non Checked In
    <br />
    <button
      className="div-seatmap-btn-mark"
      disabled={true}
      style={
        Object {
          "background": "#0080ff",
        }
      }
    />
     : 
    Wheel Chaired Passengers
     (Checked In)
    <br />
    <button
      className="div-seatmap-btn-mark"
      disabled={true}
      style={
        Object {
          "background": "#f5504e",
        }
      }
    />
     : 
    Passengers With Children
     (Checked In)
    <br />
    <button
      className="div-seatmap-btn-mark"
      disabled={true}
      style={
        Object {
          "background": "#686868",
        }
      }
    />
     : 
    Seats Not Booked
    <br />
    <br />
    <em
      style={
        Object {
          "color": "red",
        }
      }
    >
       * Click the seats to make Check In and Check out
    </em>
  </div>
  <div
    className="div-seatmap-textcontent"
  />
</div>
`;
